src/index.js

```diff
@@ -118,43 +144,40 @@ class IOpipeWrapperClass {
       s: stringValue
     });
   }
-  invoke(originalEvent, originalContext, originalCallback) {
-    this.invokeSetup(originalEvent, originalContext, originalCallback);
-    try {
-      return this.userFunc.call(
-        this,
-        this.originalEvent,
-        this.modifiedContext,
-        this.modifiedCallback
-      );
-    } catch (err) {
-      return this.sendReport(err);
-    }
-  }
 }
 
 module.exports = options => {
   const config = setConfig(options);
-  const fn = userFunc => {
+  const dnsPromise = getDnsPromise(config.host);
+  const libFn = userFunc => {
     if (!config.clientId) {
       // No-op if user doesn't set an IOpipe token.
       return userFunc;
     }
 
-    const dnsPromise = getDnsPromise(options.host);
+    // Assign .log (deprecated) here to avoid type errors
+    if (typeof libFn.log !== 'function') {
+      libFn.log = () => {};
+    }
 
-    const wrapper = new IOpipeWrapperClass(dnsPromise, config, userFunc);
-    fn.log = (...args) => {
-      console.warn(
-        'iopipe.log is deprecated and will be dropped in a future version, please use context.iopipe.log'
+    return (originalEvent, originalContext, originalCallback) => {
+      return new IOpipeWrapperClass(
+        libFn,
+        dnsPromise,
+        config,
+        userFunc,
+        originalEvent,
+        originalContext,
+        originalCallback
       );
-      wrapper.log.apply(wrapper, args);
     };
-
-    return wrapper.invoke.bind(wrapper);
   };
 
   // Alias decorate to the wrapper function
-  fn.decorate = fn;
-  return fn;
+  libFn.decorate = libFn;
+  // Used for tests
+  if (process.env.DNS_ON_LIB) {
```

Comment:
this env var shouldn't be part of the public API contract. We should find another way.